src/assembly/local/quadrature_table.rs

```diff
@@ -147,6 +147,38 @@ where
             data: self.data,
         }
     }
+
+    /// Replaces the data of the quadrature table with the given data.
+    pub fn with_data<NewData>(self, data: NestedVec<NewData>) -> GeneralQuadratureTable<T, GeometryDim, NewData> {
+        GeneralQuadratureTable {
+            points: self.points,
+            weights: self.weights,
+            data: data,
+        }
+    }
+
+    /// Replaces the data of the quadrature table by calling the given closure with every quadrature
+    /// point in reference coordinates and its element index.
+    pub fn with_data_from_fn<NewData>(
+        self,
+        mut data_fn: impl FnMut(usize, &OPoint<T, GeometryDim>, &Data) -> NewData,
+    ) -> GeneralQuadratureTable<T, GeometryDim, NewData> {
+        let mut data = NestedVec::new();
+
+        for (element_index, (points, datas)) in self.points.iter().zip(self.data.iter()).enumerate() {
+            let mut arr = data.begin_array();
+
+            for (point, data) in points.iter().zip(datas.iter()) {
+                arr.push_single(data_fn(element_index, point, data));
+            }
+        }
+
+        GeneralQuadratureTable {
+            points: self.points,
+            weights: self.weights,
+            data: data,
+        }
```

Comment:
Can maybe call `Self::with_data` here? (which will also validate after the change in the other comment)